src/main/scala/org/scalajs/tools/tsimporter/sc/Definitions.scala

```diff
@@ -334,6 +335,18 @@ object TypeRef {
   val Nothing = TypeRef(scala dot Name("Nothing"))
   val This = Singleton(QualifiedName(Name.THIS))
 
+  object Optional {
```

Comment:
This does not really make sense here. At the level of `sc.Definitions`, we are working with a Scala.js type system. There is nothing special about `js.UndefOr[A]` in the Scala.js type system (contrary to optional stuff in TypeScript). So when building a `js.UndefOr` in the `Importer`, that's where we should write `TypeRef(QualifiedName.UndefOr, List(tpe))`.

`Union` and `Intersection` are different because they have special syntactic behavior. `UndefOr` does not need this.

Reply:
Addressed in https://github.com/sjrd/scala-js-ts-importer/pull/105/commits/82b33f84c18151e9d6fecdefd29135787caa2909